### PR TITLE
Feature/add links between concepts

### PR DESCRIPTION
**Pull request by fespinoza**

### CODE REVIEW

File: LinkedIdeas/CanvasView.swift [hunk 1]
```diff
@@ -42,23 +47,126 @@ class CanvasView: NSView {
...
+    if mode == Mode.Links {
+      sprint("mouse dragged")
+      arrowEnd = convertPoint(theEvent.locationInWindow, fromView: nil)
+    }
+  }
+  
+  override func mouseUp(theEvent: NSEvent) {
+    if mode == Mode.Links {
+      sprint("mouse up")
+      if let targetConceptIdentifier = targetConceptIdentifier, originConceptIdentifier = originConceptIdentifier {
```
Comment (RamonGilabert) [inline LinkedIdeas/CanvasView.swift]: If you remove the arrows anyways, you can put `removeArrow()` outside and concatenate the two ifs, like:

```swift
if let targetConceptIdentifier = targetConceptIdentifier, originConceptIdentifier = originConceptIdentifier where  originConceptIdentifier != targetConceptIdentifier {  }
```

File: LinkedIdeas/CanvasView.swift [hunk 2]
```diff
@@ -42,23 +47,126 @@ class CanvasView: NSView {
...
+  // MARK: - Link Functions
+  
+  func removeArrow() {
+    arrowStart = nil
+    arrowEnd = nil
+  }
+  
+  func createLink(originIdentifier: Int, _ targetIdentifier: Int) {
+    let origin = concepts.filter({ element in element.identifier == originIdentifier }).first
+    let target = concepts.filter({ element in element.identifier == targetIdentifier }).first
```
Comment (RamonGilabert) [inline LinkedIdeas/CanvasView.swift]: Isn't target and origin the same thing?

File: LinkedIdeas/ConceptView.swift [hunk 3]
```diff
@@ -9,115 +9,113 @@
...
-      } else {
-        if concept.editing {
-          sprint("edit concept")
-          enableTextField()
-        } else {
-          sprint("just render it")
-          justRenderConcept()
-        }
-      }
+    if concept.editing {
```
Comment (RamonGilabert) [inline LinkedIdeas/ConceptView.swift]: concept.editing ? enableTextField() : drawConceptString()

File: LinkedIdeas/LinkView.swift [hunk 4]
```diff
@@ -0,0 +1,168 @@
+//
+//  LinkView.swift
+//  LinkedIdeas
+//
+//  Created by Felipe Espinoza Castillo on 28/11/15.
```
Comment (RamonGilabert) [inline LinkedIdeas/LinkView.swift]: :european_castle:

File: NSRect+PointHelpers.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
+//
+//  NSRect+PointHelpers.swift
+//  LinkedIdeas
+//
+//  Created by Felipe Espinoza Castillo on 04/12/15.
+//  Copyright © 2015 Felipe Espinoza Dev. All rights reserved.
+//
+
+import Cocoa
```
Comment (RamonGilabert) [inline NSRect+PointHelpers.swift]: :chocolate_bar:

File: LinkedIdeas/CanvasView.swift (hunk 1 above)
Comment (fespinoza) [inline LinkedIdeas/CanvasView.swift]: :+1:

**Pull request merged**